woocommerce/woocommerce-ios
# [Better Customer Selector] Show customer list UI

Closes: #10307 


## Description

With this PR we show a paginated searchable list of customers that can be selected (not implemented yet). Please note that the search is still based on the old version, which only searches by name.

## Changes
- Extend `SearchViewController` and `SearchUICommand` to include this task's requirements
- Add necessary functionality to `CustomerSearchUICommand` filtered by the feature flag
- Create `CustomerSelectorViewController` to contain the `SearchViewController` with loading animation
- Open the new view when tapped in the Order Details
- Unit tests
- Disable the feature flag as the new functionality is not complete. That way we don't affect development or testing.

## Testing instructions

Enable the `betterCustomerSelectionInOrder` in `DefaultFeatureFlagService`

1. Go to Orders
2. Tap on + to create a new order
3. Tap on + Add Customer Details
4. See that the list is loaded and you can search by name 

With the feature flag disabled, make sure that the old functionality works as expected

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/1864060/37ba29c9-eb47-4954-8b1e-bc76b3ac4dbc

---
